commodore/cluster.py

```diff
@@ -13,14 +13,79 @@
 )
 
 from .config import Config
-
-
-def fetch_cluster(cfg, clusterid):
-    cluster = lieutenant_query(cfg.api_url, cfg.api_token, "clusters", clusterid)
-    # TODO: move Commodore global defaults repo name into Lieutenant
-    # API/cluster facts
-    cluster["base_config"] = "commodore-defaults"
-    return cluster
+from .git import HEAD
+
+
+class Cluster:
+    def __init__(
+        self,
+        cfg: Config,
+        cluster_response: Dict,
+        tenant_response: Dict,
+    ):
+        self.__cfg = cfg
+        self.__cluster = cluster_response
+        self.__tenant = tenant_response
```

Comment:
Instead of saving the raw responses here, consider directly extracting the fields you care about in the constructor.

Reply:
I prefer to leave it that way. My first argument is: it keeps functions small. Yes, doing as you suggested would make the getters even smaller but the constructor will be bloated. Second I expect this to be simpler to extend as there is less boiler plate code to change wen new fields are accessed.